test/Interpreter/Inputs/layout_string_witnesses_types.swift

```diff
@@ -620,8 +620,8 @@ public enum MultiPayloadError {
 }
 
 public enum TwoPayloadInner {
-    case x(AnyObject)
-    case y(Int)
+    case x(Int)
+    case y(AnyObject)
```

Comment:
The reason for this change is that the bug was masked by the fact that we were using tag 0 in the test. The bug caused the first byte of the object to be read, instead of the tag byte. When this is a reference, it will always cause the tag to be read as 0, causing the test to succeed. So now we are using 1, which should make the test more reliable.

Reply:
Should we duplicate this test and verify both declaration orders?  (For the main PR only; the 6.0 cherry-pick doesn't need to wait for that.)

Reply:
For that matter, what happens with a 257-element enum?   Does that end up with a 9-bit tag?